### PR TITLE
[FW][FIX]l10n_ar_ux:recompute account move lines if l10n_ar_currency_rate

### PR DESCRIPTION
**Pull request by fw-bot-adhoc**
Ticket 50785

Forward-Port-Of: ingadhoc/odoo-argentina#604

### CODE REVIEW

**Comment by roboadhoc** [comment]
[Pull request status dashboard](https://runbot.adhoc.com.ar/ingadhoc/odoo-argentina/pull/612).

**Comment by fw-bot-adhoc** [comment]
Ping @pablohmontenegro, @zaoral cherrypicking of pull request ingadhoc/odoo-argentina#604 failed.

stdout:
```
Auto-merging l10n_ar_ux/models/account_move.py
CONFLICT (content): Merge conflict in l10n_ar_ux/models/account_move.py
Auto-merging l10n_ar_ux/__manifest__.py
CONFLICT (content): Merge conflict in l10n_ar_ux/__manifest__.py

```

stderr:
```
20:24:58.659582 git.c:439               trace: built-in: git cherry-pick 381baed2980d0941d898c01c2def717ca1cefad3
error: could not apply 381baed2... [FIX]l10n_ar_ux:recompute account move lines if l10n_ar_currency_rate
hint: after resolving the conflicts, mark the corrected paths
hint: with 'git add <paths>' or 'git rm <paths>'
hint: and commit the result with 'git commit'
----------
status:

```

Either perform the forward-port manually (and push to this branch, proceeding as usual) or close this PR (maybe?).

In the former case, you may want to edit this PR message as well.

File: l10n_ar_ux/models/account_move_line.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        for line in force_currency_rate_lines:
+            company = line.move_id.company_id
+            if line.move_id.is_invoice(include_receipts=True):
+                line.with_context(force_rate=line.move_id.l10n_ar_currency_rate)._onchange_price_subtotal()
+            elif not line.move_id.reversed_entry_id:
+                balance = line.currency_id.with_context(force_rate=line.move_id.l10n_ar_currency_rate)._convert(line.amount_currency, company.currency_id, company, line.move_id.date or fields.Date.context_today(line))
+            line.debit = balance if balance > 0.0 else 0.0
+            line.credit = -balance if balance < 0.0 else 0.0
+
+        super(AccountMoveLine, self - force_currency_rate_lines)._onchange_currency(self)
```
Comment (zaoral) [inline l10n_ar_ux/models/account_move_line.py]: remove self as argument of the call

File: l10n_ar_ux/models/__init__.py [hunk 2]
```diff
@@ -6,6 +6,7 @@
 from . import account_account_tag
 from . import account_chart_template
 from . import account_move
+from . import account_move_line
```
Comment (zaoral) [inline l10n_ar_ux/models/__init__.py:9]: please merge the two commits in one and also make a rebase with 15.0

File: l10n_ar_ux/models/account_move_line.py (hunk 1 above)
Comment (zaoral) [reply l10n_ar_ux/models/account_move_line.py]: please also review how is the code in 13.0 because maybe   we need to fix it also there

**Comment by zaoral** [comment]
@pablohmontenegro

**Comment by pablohmontenegro** [comment]
@roboadhoc r+

**Comment by pablohmontenegro** [comment]
Hi @zaoral I suppose that I need "At least 1 approving review is required to merge this pull request.". I commented "@roboadhoc r+" yesterday but the commit is not merged.

**Comment by zaoral** [comment]
@roboadhoc override=ci/argentina-ee

**Review by zaoral — APPROVED** [review]

**Pull request closed without merging**